Chauffeur/Deliverables/DeliveryDeliverable.cs

```diff
@@ -218,6 +225,34 @@ private async Task<bool> SetupDatabase()
             return true;
         }
 
+        private async Task<bool> AreAllParametersSpecified(
+            IEnumerable<string> instructions,
+            IDictionary<string, string> @params)
+        {
+            var parameters = instructions
+                .SelectMany(x => GetInstructionParametersNames(x))
+                .Distinct()
+                .OrderBy(x => x)
+                .ToList();
+
+            var missingParameters = parameters.Except(@params.Keys).OrderBy(x => x).ToList();
+            if (missingParameters.Count != 0)
+            {
+                await Out.WriteLineAsync($"The following parameters have not been specified: {string.Join(", ", missingParameters)}.");
```

Comment:
I'd actually prefer the output to be over multiple lines, makes it easier to read from a log rather than sanning a (potentially) long single line.